molecule/driver/hetznercloud.py

```diff
@@ -0,0 +1,190 @@
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import os
+
+from molecule import logger, util
+from molecule.driver import base
+from molecule.util import sysexit_with_message
+
+log = logger.get_logger(__name__)
+
+
+class HetznerCloud(base.Base):
+    """
+    The class responsible for managing `Hetzner Cloud`_ instances.
+    `Hetzner Cloud`_ is **not** the default driver used in Molecule.
+
+    Molecule leverages Ansible's `hcloud_server module`_, by mapping variables
+    from ``molecule.yml`` into ``create.yml`` and ``destroy.yml``.
+
+    .. important::
+
+        The ``hcloud_server`` module is available in the Ansible release 2.8.
+        Until that release is made, you must use the Ansible development branch
+        to get access to the module. Please see the Ansible `pip installation`_
+        documentation for further information.
+
+    .. _`hcloud_server module`: https://docs.ansible.com/ansible/devel/modules/hcloud_server_module.html#hcloud-server-module
+    .. _`pip installation`: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#latest-releases-via-pip
+    .. _`Hetzner Cloud`: https://www.hetzner.com/cloud
+
+    .. code-block:: yaml
+
+        driver:
+          name: hetznercloud
+        platforms:
+          - name: instance
+            server_type: cx11
+            ssh_keys:
+              - me@myorganisation
+            image: debian-9
+
+    .. note::
+
+        Due to the way ``hcloud_server`` makes connections, you will need to
+        ensure that the identity file for the SSH key you reference in
+        ``ssh_keys`` is loaded locally by your ``ssh-agent``. Unlike other
+        driver implementations, Molecule does not generate an SSH key for the
+        newly created instance because you must pre-register that key with your
+        Hetzner Cloud account access.
+
+    .. code-block:: bash
+
+        $ pip install 'molecule[hetznercloud]'
+
+    Change the options passed to the ssh client.
+
+    .. code-block:: yaml
+
+        driver:
+          name: hetznercloud
+          ssh_connection_options:
+            -o ControlPath=~/.ansible/cp/%r@%h-%p
+
+    .. important::
+
+        Molecule does not merge lists, when overriding the developer must
+        provide all options.
+
+    Provide the files Molecule will preserve upon each subcommand execution.
+
+    .. code-block:: yaml
+
+        driver:
+          name: hetznercloud
+          safe_files:
+            - foo
+    """  # noqa
+
+    def __init__(self, config):
+        super(HetznerCloud, self).__init__(config)
+        self._name = 'hetznercloud'
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
```

Comment:
Why do you need a setter here? Is anything outside supposed to assign thing to the public attribute?